docs/howto/manage-storage.md

```diff
@@ -2,136 +2,68 @@
 
-Now, in the body of the charm definition, define the event handler, or adjust an existing holistic one. For example, to provide the location of the attached storage to the workload configuration:
+Next, access the storage in `_update_configuration`. Don't hard-code the location of the storage. Instead, get the location from [`self.model.storages`](ops.Model.storages):
 
-```
+```python
 def _update_configuration(self, event: ops.EventBase):
     """Update the workload configuration."""
     cache = self.model.storages["cache"]
-    if cache.location is None:
-        # This must be one of the other events. Return and wait for the storage-attached event.
+    if not cache:
+        # The storage-attached event hasn't happened yet.
         logger.info("Storage is not yet ready.")
         return
-    try:
-        self.push_configuration(cache_dir=cache.location)
-    except ops.pebble.ConnectionError:
-        # Pebble isn't ready yet. Return and wait for the pebble-ready event.
-        logger.info("Pebble is not yet ready.")
-        return
+    # Storage is available. Write some data to the cache.
+    cache_root = pathops.LocalPath(cache[0].location)
+    (cache_root / "data.json").write_text("...")
 ```
 
-> Examples: [ZooKeeper ensuring that permission and ownership is correct](https://github.com/canonical/zookeeper-operator/blob/106f9c2cd9408a172b0e93f741d8c9f860c4c38e/src/charm.py#L247), [Kafka configuring additional storage](https://github.com/canonical/kafka-k8s-operator/blob/25cc5dd87bc2246c38fc511ac9c52f35f75f6513/src/charm.py#L298)
+This example uses {external+charmlibs:ref}`charmlibs-pathops <charmlibs-pathops>` to write data to the storage.
 
-### Observe the detaching event and define an event handler
+Instead of directly writing data to the storage, your charm could put the storage location in a configuration file. For example:
 
-In the `src/charm.py` file, in the `__init__` function of your charm, set up an observer for the detaching event associated with your storage and pair that with an event handler. For example:
-
-```
-self.framework.observe(self.on.cache_storage_detaching, self._on_storage_detaching)
-```
-
-> See more: [](ops.StorageDetachingEvent)
-
-Now, in the body of the charm definition, define the event handler, or adjust an existing holistic one. For example, to warn users that data won't be cached:
-
-```
-def _on_storage_detaching(self, event: ops.StorageDetachingEvent):
-    """Handle the storage being detached."""
-    self.unit.status = ops.ActiveStatus("Caching disabled; provide storage to boost performance)
+```python
+    ...
+    # Storage is available. Pass the storage location to the workload.
+    config_path = pathops.LocalPath("/etc/my-app/cache-path.config")
+    config_path.write_text(cache[0].location)
```

Comment:
And perhaps we could use all the storages here too, to make it clear that it's not always `[0]`. Something like:

```suggestion
    config_path = pathops.LocalPath("/etc/my-app/cache.config")
    config_path.write_text(f"paths = { ','.join(s.location for s in cache) }")
```